http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesClusterComponent.scala

```diff
@@ -0,0 +1,262 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import java.time.Instant
+
+import org.broadinstitute.dsde.workbench.google2.GKEModels.KubernetesClusterName
+import org.broadinstitute.dsde.workbench.google2.KubernetesModels.KubernetesApiServerIp
+import org.broadinstitute.dsde.workbench.google2.{Location, NetworkName, SubnetworkName}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import slick.lifted.Tag
+import LeoProfile.api._
+import LeoProfile.mappedColumnImplicits._
+import cats.data.Chain
+import cats.implicits._
+import nodepoolQuery.unmarshalNodepool
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.KubernetesNamespaceName
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.{dummyDate, unmarshalDestroyedDate}
+
+import scala.concurrent.ExecutionContext
+
+case class KubernetesClusterRecord(id: KubernetesClusterLeoId,
+                                   googleProject: GoogleProject,
+                                   clusterName: KubernetesClusterName,
+                                   // the GKE API actually supports a location (e.g. us-central1 or us-central1-a)
```

Comment:
This is a nice comment -- can we put it in `core/kubernetesModels` which is probably where people will mostly look?

Reply:
Agreed. Reading this comment also made me wonder if we should reflect the fact that a `Location` can be a `Region` or a `Zone` in our workbench-libs modeling of it so it's more explicit. I remember being thrown off when I first did anything with it.

Reply:
Yeah... next pr to wb-libs we should fix it, and make some sort of hierarchy where Location is an interface and `Zone` and `Region` inherit from it. Currently, `Location` already exists as an ADT in some experimental service file. 

I held off on it because it falls under a larger category that I want to address of 'stray google2 models that are applicable to more things than the Service/Interp where they are located'